pkg/server/container_create.go

```diff
@@ -222,6 +222,7 @@ func (c *criService) CreateContainer(ctx context.Context, r *runtime.CreateConta
 	// the spec for us.
 	userstr, err := generateUserString(
 		securityContext.GetRunAsUsername(),
+		image.ImageSpec.Config.User,
```

Comment:
Can we use the `getUserFromImage` helper?
And only pass in uid and username when CRI hasn't specified them.

Reply:
I believe the latest commit is what you want. And we cant use `getUserFromImage` because we need the "full" name which could be `uid:gid` etc. The parsing will happen in `oci.WithUser` so we dont want to remove the context before passing.

Reply:
SG.